controllers/openstackdataplanenodeset_controller.go

```diff
@@ -338,6 +331,25 @@ func (r *OpenStackDataPlaneNodeSetReconciler) SetupWithManager(mgr ctrl.Manager)
 		return nil
 	})
 
+	deploymentWatcher := handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		var namespace string = obj.GetNamespace()
+		var nodeSetName string = obj.GetName()
```

Comment:
Actually here the  `obj` is `OpenStackDataPlaneDeployment` object  and not the `OpenStackDataPlaneNodeSet`. We should fetch the nodesets  matching the names in obj.Spec.NodeSets to build reconcile requests for them. At present it works because the OpenStackDataPlaneDeployment and OpenStackDataPlaneNodeSet names are the same.

Reply:
Thanks, I misunderstood that initially. Should be fixed now.